src/sweeps/bayes_search.py

```diff
@@ -249,6 +243,10 @@ def next_sample(
 
     filtered_X, filtered_y = filter_nans(sample_X, sample_y)
 
+    warnings = ""
+    if sample_y.shape[0] == 0:
```

Comment:
this isn't the right condition to filter on; by this point the y values have already been imputed with zero if they are missing a value of the metric. we should do this check in `_construct_gp_data`.